src/main/java/seedu/sellsavvy/logic/commands/ordercommands/UnmarkOrderCommand.java

```diff
@@ -0,0 +1,108 @@
+package seedu.sellsavvy.logic.commands.ordercommands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.sellsavvy.logic.Messages.MESSAGE_ORDERLIST_DOES_NOT_EXIST;
+
+import seedu.sellsavvy.commons.core.index.Index;
+import seedu.sellsavvy.commons.util.ToStringBuilder;
+import seedu.sellsavvy.logic.Messages;
+import seedu.sellsavvy.logic.commands.Command;
+import seedu.sellsavvy.logic.commands.CommandResult;
+import seedu.sellsavvy.logic.commands.exceptions.CommandException;
+import seedu.sellsavvy.model.Model;
+import seedu.sellsavvy.model.order.Count;
+import seedu.sellsavvy.model.order.Date;
+import seedu.sellsavvy.model.order.Item;
+import seedu.sellsavvy.model.order.Order;
+import seedu.sellsavvy.model.order.OrderList;
+import seedu.sellsavvy.model.order.Status;
+import seedu.sellsavvy.model.person.Person;
+
+/**
+ * Reverts an order to the pending status.
+ */
+public class UnmarkOrderCommand extends Command {
+
+    public static final String COMMAND_WORD = "unmarkOrder";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Reverts an order to the pending status."
+            + "Parameters: ORDER_INDEX (must be positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 2";
+
+    public static final String MESSAGE_UNMARK_ORDER_SUCCESS =
+            "The order has been reverted to the pending status: %1$s";
+    public static final String MESSAGE_ORDER_ALREADY_UNMARKED_WARNING = "Note: "
+            + "This order is already in the pending status"
+            + ", verify if this is a mistake\n";
+
+    private final Index index;
+
+    /**
+     * Creates a UnmarkOrderCommand to revert the order of the
+     * specified index to the pending status.
+     *
+     * @param index of the order in the displayed order list to mark.
+     */
+    public UnmarkOrderCommand(Index index) {
+        requireNonNull(index);
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Person selectedPerson = model.getSelectedPerson().get();
+        if (selectedPerson == null) {
+            throw new CommandException(MESSAGE_ORDERLIST_DOES_NOT_EXIST);
+        }
+
+        OrderList orderList = selectedPerson.getOrderList();
```

Comment:
it shouldn't be the class Order List. size but the filtered list . size ( I told u if u pr after kx u have to fix it yourself)

Reply:
Also in line 54, likely you have to refactor that as well so just take note.